src/Components/Web.JS/src/Rendering/BrowserRenderer.ts

```diff
@@ -370,23 +382,28 @@ export class BrowserRenderer {
     }
 
     switch (element.tagName) {
-      case 'INPUT':
-      case 'SELECT':
-      case 'TEXTAREA': {
+      case 'INPUT': {
         const value = attributeFrame ? frameReader.attributeValue(attributeFrame) : null;
+        element[inputValuePropname] = value;
 
-        if (element instanceof HTMLSelectElement) {
-          setSelectElementValue(element, value);
+        return true;
+      }
+      case 'SELECT': {
+        const value = attributeFrame ? frameReader.attributeValue(attributeFrame) : null;
+        setSelectElementValue(element as HTMLSelectElement, value);
 
-          // <select> is special, in that anything we write to .value will be lost if there
-          // isn't yet a matching <option>. To maintain the expected behavior no matter the
-          // element insertion/update order, preserve the desired value separately so
-          // we can recover it when inserting any matching <option> or after inserting an
-          // entire markup block of descendants.
-          element[selectValuePropname] = value;
-        } else {
-          (element as any).value = value;
-        }
+        // <select> is special, in that anything we write to .value will be lost if there
+        // isn't yet a matching <option>. To maintain the expected behavior no matter the
+        // element insertion/update order, preserve the desired value separately so
+        // we can recover it when inserting any matching <option> or after inserting an
+        // entire markup block of descendants.
+        element[selectValuePropname] = value;
+
+        return true;
+      }
+      case 'TEXTAREA': {
+        const value = attributeFrame ? frameReader.attributeValue(attributeFrame) : null;
+        (element as any).value = value;
```

Comment:
Just trying to figure out if it's necessary/desirable to split up input/select/textarea into three separate code paths here. If we were willing to defer the assignment of `value` on textarea as well (and I don't know of any reason to treat it differently), could all this coalesce back into a single code path?

Rather than having `selectValuePropname` and `inputValuePropname` as distinct properties, could we have a single `deferredValuePropname` that's applied to all three element types?

I know there would still be a small special-case aspect for select (because of needing to call `setSelectElementValue`), but it might be able to share the rest of the logic with input/textarea, which wouldn't themselves need to be distinguished.

If you think it's already better as-is then that's fine. I'm just wondering if we can reduce the number of distinct cases because there's already a lot of branching in here that we have to mentally trace through each time we want to modify the behavior.

Reply:
I agree, unifying the code paths makes it a bit cleaner. I removed `setSelectElementValue` in favor of `setDeferredElementValue`, which sets the `value` property of an element whose value gets assigned after other attributes have been populated, and the special case for `select` elements happens in there (since the same check would happen in the initial assignment of the value as well as the deferred assignment). Let me know if you think this should be done differently.